### PR TITLE
[stable/airflow] 7.1.0 release

### PR DESCRIPTION
**Pull request by thesuperzapper**
This PR aims to improve upon the work already done in 7.0.0, fixing some issues and implementing better support for `KubernetesExecutor`.

It also adds some example `custom-values.yaml` for people to start from. For now, they are only for minikube and google-cloud.

For a full list of changes, please review the new `CHANGELOG.md`

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by thesuperzapper** [comment]
@gsemet  can you please review?

File: stable/airflow/README.md [hunk 1]
```diff
@@ -658,163 +698,4 @@ __Airflow Prometheus Values:__
...
-If you were using them, you'll have to migrate your settings to `postgres.existingSecret`, `redis.existingSecret`, and `airflow.extraEnv`, which are described in greater depth in the documentation above.
-
-### To 2.8.3
-The parameter `airflow.service.type` no longer applies to the Flower service, but the default of `ClusterIP` has been maintained.  
-If using a custom values file and have changed the service type, also specify `flower.service.type`.
-
-### To 2.0.0
-The parameter `workers.pod.annotations` has been renamed to `workers.podAnnotations`.  
-If using a custom values file, rename this parameter.
+| `prometheusRule.groups` | alerting rules for Prometheus | `[]` |
```
Comment (gsemet) [inline stable/airflow/README.md:706]: Can you keep the "Upgrading" section and only add a link to Changelog.md?

File: stable/airflow/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
+# Changelog
```
Comment (gsemet) [inline stable/airflow/CHANGELOG.md]: This is not really a changelog actually, this is an "upgrading guide", with what changed between 2 releases. Most of the time the minor changes / bug fixes are not described here, and I do not know if we want it.


I think it should be important to have clearly 2 types of informations:
- Changelog with the details of what changed (bug fixes....)
- upgrading information with only the changes that impact users (added/removed/moved/changed values in configuration,...)

My reference is the Home Assistant Release Blog, where they follow this structure at each major releases:
- human description of the purpose of the release
- upgrade guide
- breaking changes
- list of all changes (this section evolves with every minor releases they do)

Could be a good idea. Do you know if there are other charts that do this?

**Comment by gsemet** [comment]
Hi. Thanks for your PL. I am not against your changes, just a few questions:
- why are you naming custom-values.yaml, while it is most of the time "values.yaml"
- moving the upgrading section to changelog might have side effect, I added comments in your code
- other stuff are fine :) can't wait to integrate them !

File: stable/airflow/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+## v7.1.0
+
+The following IMPROVEMENTS have been made:
+* We have dramatically reduced the start time of airflow pods.
+  This was mostly achieved by removing arbitrary delays in the start commands for airflow pods.
```
Comment (gsemet) [inline stable/airflow/CHANGELOG.md]: I do not remember why they were needed, do you know?

**Review by gsemet — CHANGES_REQUESTED** [review]

File: stable/airflow/CHANGELOG.md (hunk 3 above)
Comment (thesuperzapper) [reply stable/airflow/CHANGELOG.md]: I think it was a bad attempt at forcing the pods to wait for dependencies, but fundamentally, thats not how K8S works, when we can just have them retry until the dependency is available, as not to increase the time under normal operation.

File: stable/airflow/templates/_helpers.tpl [hunk 4]
```diff
@@ -73,11 +73,11 @@ Construct the AIRFLOW__CORE__SQL_ALCHEMY_CONN connection string.
 */}}
 {{- define "airflow.connection.alchemy" -}}
 {{- if .Values.postgresql.enabled -}}
-postgresql+psycopg2://${DATABASE_USER}:$( python -c "import urllib.parse; encoded_pass = urllib.parse.quote('''${DATABASE_PASSWORD}'''); print(encoded_pass)" )@${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_DB}
+postgresql+psycopg2://${DATABASE_USER}:$( echo ${DATABASE_PASSWORD} | python3 -c "import urllib.parse; encoded_pass = urllib.parse.quote(input()); print(encoded_pass)" )@${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_DB}
```
Comment (gsemet) [inline stable/airflow/templates/_helpers.tpl:76]: how does this work with the `input()` ?

Comment (thesuperzapper) [reply stable/airflow/templates/_helpers.tpl:76]: Its a pipe?

**Comment by gsemet** [comment]
ok for me, let's merge

**Comment by thesuperzapper** [comment]
Yep, just finished up with some fixes to the documentation formatting.

File: stable/airflow/README.md (hunk 1 above)
Comment (thesuperzapper) [reply stable/airflow/README.md:706]: done

**Pull request merged**